GDWeave/Script/ModUtil.cs

```diff
@@ -10,6 +10,58 @@ public interface IWaiter {
     public bool Check(Token token);
 }
 
+public class FunctionWaiter : IWaiter {
+    private MultiTokenWaiter _FunctionWaiter = new([
+        t => t.Type == TokenType.PrFunction,
+        t => t.Type == TokenType.Identifier,
+    ]);
+
+    public bool Matched {get; private set;} = false;
+    public bool Ready {get; private set;} = true;
+
+    private readonly string name;
+    public FunctionWaiter(string name) {
+        this.name = name;
+    }
+
+    public void Reset() {
+        foundFunction = false;
+        foundColon = false;
+    }
+
+    public void SetReady() {
+        Ready = true;
+    }
+
+    private bool foundFunction = false;
+    private bool foundColon = false;
+    public bool Check(Token token) {
+        if (foundColon && token.Type == TokenType.Newline) {
+            Reset();
+            return true;
+        }
+
+        if (foundFunction && token.Type == TokenType.Colon) {
+            foundColon = true;
+            return false;
+        }
+
+        if (!_FunctionWaiter.Check(token)) {
+            return false;
+        }
+        else {
+            _FunctionWaiter.Reset();
```

Comment:
This comment isn't a blocker for merge, but I think this control flow is a little hard to understand - I would drag the identifier check into the waiter itself like so:

```cs
new MultiTokenWaiter([
    t => t.Type == TokenType.PrFunction,
    t => t is IdentifierToken identifier && identifier.Name == name
]);
```

Feel free to make this change if you want, but it's not required!

Reply:
Smarted added in commits